### PR TITLE
Rebase to kubernetes-1.13

### PR DESCRIPTION
**Pull request by ringtail**
rebase code base to release-1.13.5

### CODE REVIEW

**Comment by ringtail** [comment]
Hold on. I have shielded some tests. And I need to recover them first.

**Comment by ringtail** [comment]
pass tests
```
➜  cluster-capacity git:(rebase-release-1.13) ✗ go test -v ./...
?   	github.com/kubernetes-incubator/cluster-capacity/cmd/cluster-capacity	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/cmd/cluster-capacity/app	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/cmd/cluster-capacity/app/options	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/cmd/genpod	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/cmd/genpod/app	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/cmd/genpod/app/options	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/cmd/hypercc	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/api	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/client	[no test files]
=== RUN   TestPrediction
W1106 17:24:53.822968   84178 authentication.go:249] No authentication-kubeconfig provided in order to lookup client-ca-file in configmap/extension-apiserver-authentication in kube-system, so client certificate authentication won't work.
W1106 17:24:53.823085   84178 authentication.go:252] No authentication-kubeconfig provided in order to lookup requestheader-client-ca-file in configmap/extension-apiserver-authentication in kube-system, so request-header client certificate authentication won't work.
W1106 17:24:53.823108   84178 authorization.go:146] No authorization-kubeconfig provided, so SubjectAccessReview of authorization tokens won't work.
--- PASS: TestPrediction (1.01s)
    simulator_test.go:233: Reason: 0, instances: &{test-node-3 3}
    simulator_test.go:233: Reason: 1, instances: &{test-node-2 2}
    simulator_test.go:233: Reason: 2, instances: &{test-node-1 1}
    simulator_test.go:236: Stop reason: &{LimitReached Maximum number of pods simulated: 6}
PASS
ok  	github.com/kubernetes-incubator/cluster-capacity/pkg/framework	1.068s
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/framework/record	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/framework/restclient/external	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/framework/store	[no test files]
=== RUN   TestAddPodStrategy
--- PASS: TestAddPodStrategy (0.00s)
PASS
ok  	github.com/kubernetes-incubator/cluster-capacity/pkg/framework/strategy	0.025s
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/framework/watch	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/test	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/pkg/utils	[no test files]
?   	github.com/kubernetes-incubator/cluster-capacity/verify/boilerplate/test	[no test files]
```
simulate scheduling.
```
➜  cluster-capacity git:(rebase-release-1.13) ✗ ./cluster-capacity --kubeconfig ~/.kube/config --podspec ../../examples/pod.yaml
I1106 17:23:40.797690   84018 defaults.go:210] TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory
41
```
simulate scheduling in verbose mode
```
➜  cluster-capacity git:(rebase-release-1.13) ✗ ./cluster-capacity --kubeconfig ~/.kube/config --podspec ../../examples/pod.yaml --verbose
I1106 17:23:53.790463   84036 defaults.go:210] TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory
small-pod pod requirements:
	- CPU: 150m
	- Memory: 100Mi

The cluster can schedule 41 instance(s) of the pod small-pod.

Termination reason: Unschedulable: 0/7 nodes are available: 2 Insufficient cpu, 5 node(s) had taints that the pod didn't tolerate.

Pod distribution among nodes:
small-pod
	- cn-beijing.192.168.3.251: 22 instance(s)
	- cn-beijing.192.168.3.250: 19 instance(s)
```

File: pkg/framework/simulator_test.go [hunk 1]
```diff
@@ -190,11 +195,23 @@ func TestPrediction(t *testing.T) {
...
-	schedServer, _ := sapps.NewSchedulerServer(ksConfig, "http://localhost:8080")
-	cc, err := New(schedServer,
+	opts, err := kubescheduleroptions.NewOptions()
+	if err != nil {
+		t.Fatalf("unable to create scheduler options: %v", err)
+	}
+
+	// inject scheduler config config
+	opts.ComponentConfig = kubeschedulerconfig.KubeSchedulerConfiguration{
+		SchedulerName: "default-scheduler",
```
Comment (ingvagabund) [inline pkg/framework/simulator_test.go]: Can you use `k8s.io/api/core/v1.DefaultSchedulerName` instead?

Comment (ringtail) [reply pkg/framework/simulator_test.go]: changed  "default-scheduler" to "k8s.io/api/core/v1".DefaultSchedulerName

File: pkg/framework/report.go [hunk 2]
```diff
@@ -107,9 +109,9 @@ func getMainFailReason(message string) *ClusterCapacityReviewScheduleFailReason
 func getResourceRequest(pod *v1.Pod) *Resources {
 	result := Resources{
 		PrimaryResources: v1.ResourceList{
-			v1.ResourceName(v1.ResourceCPU):       *resource.NewMilliQuantity(0, resource.DecimalSI),
-			v1.ResourceName(v1.ResourceMemory):    *resource.NewQuantity(0, resource.BinarySI),
-			v1.ResourceName(v1.ResourceNvidiaGPU): *resource.NewMilliQuantity(0, resource.DecimalSI),
+			v1.ResourceName(v1.ResourceCPU):    *resource.NewMilliQuantity(0, resource.DecimalSI),
+			v1.ResourceName(v1.ResourceMemory): *resource.NewQuantity(0, resource.BinarySI),
+			//v1.ResourceName(v1.ResourceNvidiaGPU): *resource.NewMilliQuantity(0, resource.DecimalSI),
```
[the remark this answers is not in the document]
Comment (ringtail) [reply pkg/framework/report.go]: Add ResourceNvidiaGPU support

File: pkg/framework/report.go [hunk 3]
```diff
@@ -25,14 +25,16 @@ import (
 
 	"github.com/ghodss/yaml"
 	"k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/kubernetes/pkg/api/unversioned"
+	//"k8s.io/kubernetes/pkg/api/unversioned"
```
[the remark this answers is not in the document]
Comment (ringtail) [reply pkg/framework/report.go]: remove unused import packages.

**Comment by ingvagabund** [comment]
```
$ minikube version
minikube version: v1.1.0
$ sudo kubectl version
...
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

```
$ ./cluster-capacity --kubeconfig ~/.kube/config --podspec examples/pod.yaml --verbose
I1113 13:05:17.997183   22250 defaults.go:210] TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory
small-pod pod requirements:
	- CPU: 150m
	- Memory: 100Mi

The cluster can schedule 8 instance(s) of the pod small-pod.

Termination reason: Unschedulable: 0/1 nodes are available: 1 Insufficient cpu.

Pod distribution among nodes:
small-pod
	- minikube: 8 instance(s)
```

**Pull request merged**